ipywidgets/widgets/widget_int.py

```diff
@@ -258,13 +259,50 @@ def _validate_bounds(self, proposal):
             self.value = (max(new, self.value[0]), max(new, self.value[1]))
         if trait.name == 'max':
             self.value = (min(new, self.value[0]), min(new, self.value[1]))
+        if self.min_gap > 0:
+            # make sure new max/min range is > min_gap
+            value_gap = self.value[1] - self.value[0]
+            range_gap = self.max - self.min
+            if range_gap < self.min_gap:
+                raise TraitError('new min/max violates min_gap')
+            if value_gap < self.min_gap:
+                # if value gap is < min_gap, revert to min or max
+                if trait.name == 'min':
+                    self.value = (self.value[0], self.max)
+                if trait.name == 'max':
+                    self.value = (self.min, self.value[1])
+                value_gap = self.value[1] - self.value[0]
+                # fallback, we shouldn't reach this point,
+                # but if we do, reset values to min/max
+                if value_gap < self.min_gap:
+                    self.value = (self.min, self.max)
         return new
 
+        @validate('min_gap')
+        def _validate_min_gap(self, proposal):
+            new_gap = proposal['value']
+            if new_gap > 0:
+                value_gap = self.value[1] - self.value[0]
+                range_gap = self.max - self.min
+                if range_gap < new_gap:
+                    raise TraitError('new min_gap violates min/max range')
+                if value_gap < range_gap:
+                    raise TraitError('new min_gap violates gap between values')
+            return new_gap
+
+
     @validate('value')
     def _validate_value(self, proposal):
         lower, upper = super(_BoundedIntRange, self)._validate_value(proposal)
         lower, upper = min(lower, self.max), min(upper, self.max)
         lower, upper = max(lower, self.min), max(upper, self.min)
+        if self.min_gap > 0:
+            value_gap = upper - lower
+            if value_gap < self.min_gap:
+                # if value gap is smaller than min_gap, clamp
+                # values within the min-max range of the slider
+                lower = self.min
+                upper = self.max
```

Comment:
Should we reset to the slider bounds, or should we just go back to the previous value? (I think `proposal` still contains the old `value`?)

Reply:
ah, thanks, didn't realize proposal had that info- I agree, resetting to the previous value would cleaner here, I'll test/try to make that change!